components/org.wso2.carbon.database.utils/src/main/java/org/wso2/carbon/database/utils/jdbc/JdbcTemplate.java

```diff
@@ -367,6 +367,29 @@ public <T extends Object> int executeBatchInsert(String query, QueryFilter query
         return 0;
     }
 
+    /**
+     * Executes the jdbc batch update query and returns the results.
+     *
+     * @param query       The SQL for update
+     * @param queryFilter Query filter to prepared statement parameter binding.
+     * @return the results of the batch update.
+     */
+    public int[] executeBatchUpdateWithResults(String query, QueryFilter queryFilter) throws DataAccessException {
```

Comment:
Shall we use the function name as `executeBatchUpdate`?

Reply:
We already have methods such as executeBatchInsert and executeBatchDelete which doesnt return results.

And for a update we have a method named `executeUpdateWithAffectedRows`

Due to above two reasons, I named the function as executeBatchUpdateWithResults

WDYT?

Reply:
Noted!
Shall we use the `...WithAffectedRows` part following the same name pattern?

Reply:
Updated the PR